src/ruby/pb/test/xds_client.rb

```diff
@@ -159,23 +180,37 @@ def get_client_stats(req, _call)
       rpcs_by_method: rpcs_by_method,
       rpcs_by_peer: watcher['rpcs_by_peer'],
       num_failures: watcher['no_remote_peer'] + watcher['rpcs_needed']
-    );
+    )
   end
 
   def get_client_accumulated_stats(req, _call)
     $accumulated_stats_mu.synchronize do
+      all_stats_per_method = {}
+      $accumulated_method_stats.each do |rpc, stats_per_method|
+        one_stats_per_method = LoadBalancerAccumulatedStatsResponse::MethodStats.new(
+          rpcs_started: stats_per_method.rpcs_started,
+          result: stats_per_method.result
+        )
+        all_stats_per_method[rpc] = one_stats_per_method
+      end
       LoadBalancerAccumulatedStatsResponse.new(
         num_rpcs_started_by_method: $num_rpcs_started_by_method,
         num_rpcs_succeeded_by_method: $num_rpcs_succeeded_by_method,
-        num_rpcs_failed_by_method: $num_rpcs_failed_by_method
+        num_rpcs_failed_by_method: $num_rpcs_failed_by_method,
+        stats_per_method: all_stats_per_method,
       )
     end
   end
 end
 
+def add_stats_per_method(rpc_stats_key, status_code)
```

Comment:
nit: since this is a very thin helper function, for readability I think it might be more clear to just inline

```
  $accumulated_method_stats[rpc_stats_key].add_result(status_code)
```

everywhere

Reply:
Comment addressed.